### PR TITLE
Tensorboard: Add histogram callback

### PR DESCRIPTION
**Pull request by YujiOshima**
## Description ##
Add visualize histograms and distributions of each layer in Tensorboard.
You can use `contrib.tensorboard.LogMetricsCallback.node_histogram_visualization` like `callback.module_checkpoint` .
You can specify the names of nodes ( e.g. fc1_bias, default is visualizing all nodes)

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: python/mxnet/contrib/tensorboard.py [hunk 1]
```diff
@@ -71,3 +72,43 @@ def __call__(self, param):
             if self.prefix is not None:
                 name = '%s-%s' % (self.prefix, name)
             self.summary_writer.add_scalar(name, value)
+
+    def node_histogram_visualization(self, prefix=None, node_names=None, bins="auto"):
```
Comment (piiswrong) [inline python/mxnet/contrib/tensorboard.py]: How is this supposed to be used? Should it be a new class?

Comment (YujiOshima) [reply python/mxnet/contrib/tensorboard.py]: Thank you for comment.
This method can be used like below.

```python
lm = mx.contrib.tensorboard.LogMetricsCallback("/log")
model.fit(train,
...
    epoch_end_callback=lm.node_histogram_visualization(),
...
)
```

I understand it would be nice that histogram callback is defined as a new class.
But if you use both of scalar and histogram callback, you should use the same `tensorboard.SummaryWriter` instance.
Another way, make a new class and give SummaryWriter instance as an argument like below.

```python
class HistogramCallback(object):
    def __init__(self, logging_dir, prefix=None, summarywriter=None):
```

Any ideas?

**Comment by CodingCat** [comment]
Hi, the community has passed to vote about associating the code changes with JIRA (https://lists.apache.org/thread.html/ab22cf0e35f1bce2c3bf3bec2bc5b85a9583a3fe7fd56ba1bbade55f@%3Cdev.mxnet.apache.org%3E)

We have updated the guidelines for contributors in https://cwiki.apache.org/confluence/display/MXNET/Development+Process, please ensure that you have created a JIRA at https://issues.apache.org/jira/projects/MXNET/issues/ to describe your work in this pull request and include the JIRA title in your PR as [MXNET-xxxx] your title where MXNET-xxxx is the JIRA id

File: python/mxnet/contrib/tensorboard.py [hunk 2]
```diff
@@ -71,3 +72,43 @@ def __call__(self, param):
...
+            for k, v in arg.items():
+                if self.node_names is None or k in self.node_names:
+                    if self.histogram_prefix is not None:
+                        name = '%s-%s' % (self.histogram_prefix, k)
+                    self.summary_writer.add_histogram(name, v, global_step=iter_no, bins=self.bins)
+            for k, v in aux.items():
+                if self.node_names is None or k in self.node_names:
+                    if self.histogram_prefix is not None:
+                        name = '%s-%s' % (self.histogram_prefix, k)
+                    self.summary_writer.add_histogram(name, v, global_step=iter_no, bins=self.bins)
```
Comment (rahul003) [inline python/mxnet/contrib/tensorboard.py]: This PR is still helpful to use Symbolic mode with mxboard. I'm trying to use it, however this name is used without being initialized
